pykokkos/lib/ufuncs.py

```diff
@@ -1381,6 +1451,147 @@ def exp2(view):
+    out[tid] = 0
+    for i in range(view.extent(1)):
+        out[tid] += (pow(view[tid][i] - view_mean[tid], 2)) / view.extent(1)
+
+
+def var(view, axis):
+    if str(view.dtype) == "DataType.double":
+        if axis == 0:
+            view_mean = mean(view, 0)
+            out = pk.View([view.shape[1]], pk.double)
+            pk.parallel_for(view.shape[1], var_2d_axis0_impl, view=view, view_mean=view_mean, out=out)
+            return out
+        else:
+            view_mean = mean(view, 1)
+            out = pk.View([view.shape[0]], pk.double)
+            pk.parallel_for(view.shape[0], var_2d_axis1_impl, view=view, view_mean=view_mean, out=out)
+            return out
+
+
+@pk.workunit
+def mean_axis0_impl_1d_double(tid: int, view: pk.View2D[pk.double], out: pk.View1D[pk.double]):
+    out[tid] = 0
+    for i in range(view.extent(0)):
+        out[tid] += (view[i][tid] / view.extent(0))
+
+
+@pk.workunit
+def mean_axis1_impl_1d_double(tid: int, view: pk.View2D[pk.double], out: pk.View1D[pk.double]):
+    out[tid] = 0
+    for i in range(view.extent(1)):
+        out[tid] += (view[tid][i] / view.extent(1))
+
+
+def mean(view, axis):
+    if str(view.dtype) == "DataType.double":
+        if axis == 0:
+            out = pk.View([view.shape[1]], pk.double)
+            pk.parallel_for(view.shape[1], mean_axis0_impl_1d_double, view=view, out=out)
+            return out
+        else:
+            out = pk.View([view.shape[0]], pk.double)
+            pk.parallel_for(view.shape[0], mean_axis1_impl_1d_double, view=view, out=out)
+
+            return out
+    else:
+        raise RuntimeError("Incompatible Types")
+
+
+@pk.workunit
+def in1d_impl_1d_double(tid: int, viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double], out: pk.View1D[pk.int8]):
+    out[tid] = 0
+    for i in range(viewB.extent(0)):
+        if viewB[i] == viewA[tid]:
+            out[tid] = 1
+            break
+
+def in1d(viewA, viewB):
+    if str(viewA.dtype) == "DataType.double":
+        out = pk.View(viewA.shape, pk.int8)
+        pk.parallel_for(
+            viewA.shape[0],
+            in1d_impl_1d_double,
+            viewA=viewA,
+            viewB=viewB,
+            out=out)
+    else:
+        raise RuntimeError("Incompatible Types")
+
+    return out
+
+
+@pk.workunit
+def transpose_impl_2d(tid: int, view: pk.View2D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(view.extent(1)):
+        out[i][tid] = view[tid][i]
+
+
+def transpose(view):
+    view_temp = pk.View(view.shape, dtype=pk.double)
+    view_temp[:] = view
+    view = view_temp
+
+    if view.rank() == 2:
+        if str(view.dtype) == "DataType.double":
+            out = pk.View(view.shape[::-1], pk.double)
+            pk.parallel_for(view.shape[0], transpose_impl_2d, view=view, out=out)
+            return out
+    
+    raise RuntimeError("Incompatible Types")
```

Comment:
Replace the error message with something related to the view's shape (not being rank 2)